### PR TITLE
Calculator page object

### PR DESCRIPTION
**Pull request by kate-krasnoselska**
Hi! Please check my work. I rewrote Calculator Page tests to use Page Object

### CODE REVIEW

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs [hunk 1]
```diff
@@ -56,116 +62,97 @@ public void AfterEachTest()
         [Test]
         public void PositiveTestCalculator365days()
         {
-            browser.FindElement(By.XPath("//td[2]//input[@id = 'amount']")).SendKeys("100");
-            browser.FindElement(By.XPath("//input [@id = 'percent']")).SendKeys("10");
-            browser.FindElement(By.XPath("//input [@id = 'term']")).SendKeys("365");
-            browser.FindElement(By.XPath("//input[@type][2]")).Click();
+            new CalculatorPage(browser)
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: If you use the same page few times, create a variable for this page and use it.

```suggestion
            CalculatorPage calculatorPage = new CalculatorPage(browser)
            calculatorPage
```

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs [hunk 2]
```diff
@@ -56,116 +62,97 @@ public void AfterEachTest()
...
+                .EnterCalculatorData365days("100", "10", "365");
 
-            //new WebDriverWait(browser, TimeSpan.FromSeconds(10))
-            //.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(By.XPath("//button [@id = 'calculateBtn']")));
+            new CalculatorPage(browser).Calculate();
 
-            //browser.FindElement(By.XPath("//button [@id = 'calculateBtn']")).Click();
-            //Thread.Sleep(1000);
+            Thread.Sleep(1000);
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: You should move this 'pause' to the 'Calculate' method.

File: Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs [hunk 3]
```diff
@@ -14,15 +14,62 @@ public CalculatorPage(IWebDriver driver)
...
+
+        public IWebElement Income => driver.FindElement(By.XPath("//input [@id = 'income']"));
+
+        public IWebElement InterestEarned => driver.FindElement(By.XPath("//input [@id = 'interest']"));
+
+        public IWebElement EndDate => driver.FindElement(By.XPath("//input [@id = 'endDate']"));
+
+        public IWebElement InterestEarnedLayout => driver.FindElement(By.XPath("//th [contains(text(), 'Interest earned')]"));
+
+        public IWebElement ActualIncome => driver.FindElement(By.XPath("//input [@id = 'income']"));
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs]: It's better to change this property type to 'string' and return the text value. This works if you don't need the control itself, only its text.
```suggestion
        public string ActualIncome => driver.FindElement(By.XPath("//input [@id = 'income']")).GetAttribute("value");
```

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs [hunk 4]
```diff
@@ -56,116 +62,97 @@ public void AfterEachTest()
...
-            browser.FindElement(By.XPath("//input [@id = 'percent']")).SendKeys("10");
-            browser.FindElement(By.XPath("//input [@id = 'term']")).SendKeys("360");
-            browser.FindElement(By.XPath("//input[@type][1]")).Click();
-
-            //browser.FindElement(By.XPath("//button [@id = 'calculateBtn']")).Click();
-            //Thread.Sleep(1000);
-
-            //CalculatorPage calculatorPage = new CalculatorPage(browser);
-            //calculatorPage.Calculate();
+            new CalculatorPage(browser)
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: Too many 'new CalculatorPage' calls in this test. Please create a variable 'calculatorPage' and use it.

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs [hunk 5]
```diff
@@ -56,116 +62,97 @@ public void AfterEachTest()
...
-            browser.FindElement(By.XPath("//td[2]//input[@id = 'amount']")).SendKeys(" ");
-            browser.FindElement(By.XPath("//input [@id = 'percent']")).SendKeys("10");
-            browser.FindElement(By.XPath("//input [@id = 'term']")).SendKeys("365");
-            browser.FindElement(By.XPath("//input[@type][2]")).Click();
-
             CalculatorPage calculatorPage = new CalculatorPage(browser);
-            calculatorPage.CalculateBtn.GetAttribute("value");
 
-            Assert.AreEqual("CalculateBtn", calculatorPage.CalculateBtn);
+            new CalculatorPage(browser)
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/CalculatorTests.cs:121]:
```suggestion
            calculatorPage
```

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs [hunk 6]
```diff
@@ -56,116 +62,97 @@ public void AfterEachTest()
...
         {
-            browser.FindElement(By.XPath("//td[2]//input[@id = 'amount']")).SendKeys("100");
-            browser.FindElement(By.XPath("//input [@id = 'percent']")).SendKeys(" ");
-            browser.FindElement(By.XPath("//input [@id = 'term']")).SendKeys("365");
-            browser.FindElement(By.XPath("//input[@type][2]")).Click();
-            string actualIncome = browser.FindElement(By.XPath("//input [@id = 'income']")).GetAttribute("value");
+            CalculatorPage calculatorPage = new CalculatorPage(browser);
 
-            browser.FindElement(By.XPath("//button [@id = 'calculateBtn']")).Click();
+            new CalculatorPage(browser)
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/CalculatorTests.cs:132]:
```suggestion
            calculatorPage
```

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs [hunk 7]
```diff
@@ -56,116 +62,97 @@ public void AfterEachTest()
...
+                .EnterCalculatorData365days("100", " ", "365");
 
-            Thread.Sleep(1000);
-            Assert.AreEqual("0.00", actualIncome);
-            string actualInterest = browser.FindElement(By.XPath("//input [@id = 'interest']")).GetAttribute("value");
-            Assert.AreEqual("0.00", actualInterest);
+            Assert.AreEqual(false, calculatorPage.CalculateBtn.Enabled);
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: For boolean checks you should use 'IsTrue' or 'IsFalse'

```suggestion
            Assert.IsFalse(calculatorPage.CalculateBtn.Enabled);
```

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs [hunk 8]
```diff
@@ -56,116 +62,97 @@ public void AfterEachTest()
...
         {
-            browser.FindElement(By.XPath("//td[2]//input[@id = 'amount']")).SendKeys("100");
-            browser.FindElement(By.XPath("//input [@id = 'percent']")).SendKeys("10");
-            browser.FindElement(By.XPath("//input [@id = 'term']")).SendKeys(" ");
-            browser.FindElement(By.XPath("//input[@type][2]")).Click();
-            browser.FindElement(By.XPath("//button [@id = 'calculateBtn']")).Click();
-            Thread.Sleep(1000);
+            CalculatorPage calculatorPage = new CalculatorPage(browser);
+
+            new CalculatorPage(browser)
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/CalculatorTests.cs]:
```suggestion
            calculatorPage
```

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs [hunk 9]
```diff
@@ -56,116 +62,97 @@ public void AfterEachTest()
...
+                .EnterCalculatorData365days("100", "10", " ");
 
-            string actualIncome = browser.FindElement(By.XPath("//input [@id = 'income']")).GetAttribute("value");
-            
-            Assert.AreEqual("0.00", actualIncome);
-            string actualInterest = browser.FindElement(By.XPath("//input [@id = 'interest']")).GetAttribute("value");
-            Assert.AreEqual("0.00", actualInterest);
+            Assert.AreEqual(false, calculatorPage.CalculateBtn.Enabled);
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/CalculatorTests.cs]:
```suggestion
            Assert.IsFalse(calculatorPage.CalculateBtn.Enabled);
```

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs [hunk 10]
```diff
@@ -56,116 +62,97 @@ public void AfterEachTest()
...
         {
-            browser.FindElement(By.XPath("//td[2]//input[@id = 'amount']")).SendKeys("100");
-            browser.FindElement(By.XPath("//input [@id = 'percent']")).SendKeys("10");
-            browser.FindElement(By.XPath("//input [@id = 'term']")).SendKeys("365");
+            CalculatorPage calculatorPage = new CalculatorPage(browser);
+
+            new CalculatorPage(browser)
+                .EnterCalculatorData365days("100", "10", "365");
 
             SelectElement daySelect = new SelectElement(element: browser.FindElement(By.XPath("//select [@id = 'day']")));
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: Note: no 'browser.FindElement' are allowed in tests. Please move this to appropriate page as control or method. Also, we can do it on the next call.

File: Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs [hunk 11]
```diff
@@ -14,15 +14,62 @@ public CalculatorPage(IWebDriver driver)
...
 
         public void Calculate()
         {
             CalculateBtn.Click();
-            Thread.Sleep(1000);
+            new WebDriverWait(driver, TimeSpan.FromSeconds(10))
+            .Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(CalculateBtnLocator));
+            Thread.Sleep(200);
+        }
+        public IWebElement DepositAmountFld => driver.FindElement(By.XPath("//td[2]//input[@id = 'amount']"));
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs]:
```suggestion
        public IWebElement DepositAmountFld => driver.FindElement(By.XPath("//input[@id = 'amount']"));
```

File: Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs [hunk 12]
```diff
@@ -14,15 +14,62 @@ public CalculatorPage(IWebDriver driver)
...
+        public IWebElement ActualIncome => driver.FindElement(By.XPath("//input [@id = 'income']"));
+
+        public IWebElement ActualInterest => driver.FindElement(By.XPath("//input [@id = 'interest']"));
+
+        public void EnterCalculatorData365days(string deposit, string rate, string term)
+        {
+            DepositAmountFld.SendKeys(deposit);
+            RateOfInterestFld.SendKeys(rate);
+            InvestmentTermFld.SendKeys(term);
+            FinancialYearRadio365.Click();
```
Comment (varajan) [inline Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs]:
```suggestion
        public void EnterCalculatorData(string deposit, string rate, string term, bool year365)
        {
            DepositAmountFld.SendKeys(deposit);
            RateOfInterestFld.SendKeys(rate);
            InvestmentTermFld.SendKeys(term);
            if (year365)
            {
                FinancialYearRadio365.Click();
            }
            else
            {
                FinancialYearRadio360.Click();
            }
```

**Review by varajan — CHANGES_REQUESTED** [review]
Nice work, will help you to resolve the issues.

File: Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs (hunk 3 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs]: done, thanks

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs (hunk 2 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: it is already done, idk why you see it
 ` public void Calculate()
        {
            CalculateBtn.Click();
            new WebDriverWait(driver, TimeSpan.FromSeconds(10))
            .Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(CalculateBtnLocator));
            Thread.Sleep(1000);
        }`

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs (hunk 1 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/CalculatorTests.cs:65]: sorry, could not understand this, let's discuss

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs (hunk 9 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: done, thanks

File: Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs (hunk 12 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs]: wow! cool! thank you, done

File: Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs (hunk 11 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/Pages/CalculatorPage.cs]: done, thanks!

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs (hunk 10 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: ok, let's do it together

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs (hunk 7 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: done, thanks

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs (hunk 1 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: i did it as well as I could understand you, let's discuss on call

 ```
CalculatorPage calculatorPage = new CalculatorPage(browser);
            calculatorPage
                .EnterCalculatorData("100", "10", "365", true);

            calculatorPage
                .Calculate();
```

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs (hunk 4 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: did it, let's discuss

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs (hunk 5 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/CalculatorTests.cs:121]: did

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs (hunk 6 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/CalculatorTests.cs:132]: did

File: Calculator.Tests/Calculator.Tests/CalculatorTests.cs (hunk 8 above)
Comment (kate-krasnoselska) [reply Calculator.Tests/Calculator.Tests/CalculatorTests.cs]: did

**Pull request merged**